sha1-validation/Program.cs

```diff
@@ -0,0 +1,51 @@
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+
+bool sha1RsaSignatureOnLastElementInChain = false;
+
+HttpClientHandler handler = new HttpClientHandler { 
+    CheckCertificateRevocationList = true,
+    ServerCertificateCustomValidationCallback = ServerCertificateCustomValidation, 
+};
+
+using HttpClient client = new HttpClient(handler);
+
+try
+{
+    HttpResponseMessage response = await client.GetAsync("https://redhat.com");
+
+    string responseBody = await response.Content.ReadAsStringAsync();
+    Console.WriteLine($"{sha1RsaSignatureOnLastElementInChain}");
+    Console.WriteLine("PASS");
+    return 0;
+}
+catch (Exception e)
+{
+    Console.WriteLine("\nException Caught!");
+    Console.WriteLine(e);
+}
+
+Console.WriteLine("FAIL");
+return 1;
+
+handler.Dispose();
+
+client.Dispose();
+
+bool ServerCertificateCustomValidation(HttpRequestMessage requestMessage, X509Certificate2? certificate, X509Chain? chain, SslPolicyErrors sslErrors)
+{
+    foreach (var element in chain!.ChainElements)
+    {
+        var cert = element.Certificate;
+        Console.WriteLine($"{cert.SubjectName.Name} {cert.SignatureAlgorithm.FriendlyName}");
+    }
+    if ( chain.ChainElements.Last().Certificate.SignatureAlgorithm.FriendlyName != "sha1RSA" )
+    {
+        Console.WriteLine("The certificate chain that is validated is no longer using an RSA1 signature.");
+        Console.WriteLine("The test must be updated to use a different host.");
+        return false;
+    }
```

Comment:
There may be multiple chains due to redirects.
The test is good is at least one of them uses a sha1 signature on the last cert.

By moving this check here, the test now requires all chains to meet that expectation.

This should go back to setting `sha1RsaSignatureOnLastElementInChain`, and we should check `sha1RsaSignatureOnLastElementInChain` is set after the HTTP response was read.

Reply:
This is the reason the test is now failing on all CI jobs.